### DISCUSSION TITLE
Relation Between Simulation Objects and the .dot Input File

### DISCUSSION THREAD

**Discussion by kipk49** [arfc/saltproc#148] [Q&A] [answered]
Apologies if this is a bit of a dumb question, but I'm having trouble figuring out how Saltproc determines the order and connections between the objects specified in a given `_objects.json` file.

Does it use the `.dot` file to determine the way in which the material stream flows between the defined objects, or is there a default order that always takes precedence?

E.g. in the MSBR example, there are many objects defined in `msbr_objects.json` but I don't see a way in which they're linked together. Looking at `msbr.dot`, however, I see that the material flow is represented there as a flowchart.

Thanks,
Kip

**Comment by yardasol** [comment, marked as answer]
Hi @kipk49, sorry for the late response on this. The `_objects.json` file _defines_ the processing system objects and related parameters (i.e. efficiency, capacity, mass flowrate, specific elements to filter, etc.). The `.dot` file described how the processing system objects connect to each other (i.e. in the MSBR example, on line 12 of `msbr.dot`, we see that the `core_outlet` object feeds into the `sparger` object.) A good way to visualize this is using [this website](http://www.webgraphviz.com/). The graph that `msbr.dot` creates is below.
![Screenshot 2022-04-19 at 14-51-58 Webgraphviz](https://user-images.githubusercontent.com/45364492/164084686-962e2dee-a0f0-427d-a1be-1ebaf750c9eb.png)

I hope this answers your question.
